api/storage_service.cc

```diff
@@ -1207,20 +1207,21 @@ void set_snapshot(http_context& ctx, routes& r, sharded<db::snapshot_ctl>& snap_
     ss::take_snapshot.set(r, [&snap_ctl](std::unique_ptr<request> req) {
         auto tag = req->get_query_param("tag");
         auto column_families = split(req->get_query_param("cf"), ",");
+        auto sf = db::snapshot_ctl::skip_flush(!req->get_query_param("skip-flush").empty());
```

Comment:
If this correct? checking for empty? what about things like skip-flush=false?

Reply:
Yeah, from looking at https://github.com/scylladb/scylla-tools-java/blob/fd92603b9985dbdf2561c7701247264c8c997d7f/src/java/org/apache/cassandra/tools/nodetool/Snapshot.java#L63
it seems you're right that it will carry a boolean value.
Will fix.

Reply:
Fixed in v2:
```diff
diff --git a/api/storage_service.cc b/api/storage_service.cc
index f1214ee6d..02feddd10 100644
--- a/api/storage_service.cc
+++ b/api/storage_service.cc
@@ -1207,7 +1207,8 @@ void set_snapshot(http_context& ctx, routes& r, sharded<db::snapshot_ctl>& snap_
     ss::take_snapshot.set(r, [&snap_ctl](std::unique_ptr<request> req) {
         auto tag = req->get_query_param("tag");
         auto column_families = split(req->get_query_param("cf"), ",");
-        auto sf = db::snapshot_ctl::skip_flush(!req->get_query_param("skip-flush").empty());
+        auto sfopt = req->get_query_param("skip-flush");
+        auto sf = db::snapshot_ctl::skip_flush(strcasecmp(sfopt.c_str(), "true") == 0);
 
         std::vector<sstring> keynames = split(req->get_query_param("kn"), ",");
 
```